packages/core/src/mutation/create_json_mutation.ts

```diff
@@ -0,0 +1,254 @@
+  Params,
+  Data,
+  TransformedData,
+  Error,
+  BodySource = void,
+  QuerySource = void,
+  HeadersSource = void,
+  UrlSource = void,
+  DataSource = void,
+  ValidationSource = void
+>(
+  config: BaseJsonMutationConfigWithParams<
+    Params,
+    TransformedData,
+    BodySource,
+    QuerySource,
+    HeadersSource,
+    UrlSource
+  > & {
+    response: {
+      contract: Contract<unknown, Data>;
+      mapData: TwoArgsDynamicallySourcedField<
+        Data,
+        Params,
+        TransformedData,
+        DataSource
+      >;
+      validate?: Validator<TransformedData, Params, ValidationSource>;
+      status?: { expected: number | number[] };
+    };
+  }
+): Mutation<
+  Params,
+  TransformedData,
+  ApiRequestError | Error | InvalidDataError
+>;
+
+// params + no mapData
+function createJsonMutation<
+  Params,
+  Data,
+  Error,
+  BodySource = void,
+  QuerySource = void,
+  HeadersSource = void,
+  UrlSource = void,
+  ValidationSource = void
+>(
+  config: BaseJsonMutationConfigWithParams<
+    Params,
+    Data,
+    BodySource,
+    QuerySource,
+    HeadersSource,
+    UrlSource
+  > & {
+    response: {
+      contract: Contract<unknown, Data>;
+      validate?: Validator<Data, Params, ValidationSource>;
+      status?: { expected: number | number[] };
+    };
+  }
+): Mutation<Params, Data, ApiRequestError | Error | InvalidDataError>;
+
+// No params + mapData
+function createJsonMutation<
+  Data,
+  TransformedData,
+  Error,
+  BodySource = void,
+  QuerySource = void,
+  HeadersSource = void,
+  UrlSource = void,
+  DataSource = void,
+  ValidationSource = void
+>(
+  config: BaseJsonMutationConfigNoParams<
+    TransformedData,
+    BodySource,
+    QuerySource,
+    HeadersSource,
+    UrlSource
+  > & {
+    response: {
+      contract: Contract<unknown, Data>;
+      mapData: TwoArgsDynamicallySourcedField<
+        Data,
+        void,
+        TransformedData,
+        DataSource
+      >;
+      validate?: Validator<TransformedData, void, ValidationSource>;
+      status?: { expected: number | number[] };
+    };
+  }
+): Mutation<void, TransformedData, ApiRequestError | Error | InvalidDataError>;
+
+// No params + no mapData
+function createJsonMutation<
+  Data,
+  Error,
+  BodySource = void,
+  QuerySource = void,
+  HeadersSource = void,
+  UrlSource = void,
+  ValidationSource = void
+>(
+  config: BaseJsonMutationConfigNoParams<
+    Data,
+    BodySource,
+    QuerySource,
+    HeadersSource,
+    UrlSource
+  > & {
+    response: {
+      contract: Contract<unknown, Data>;
+      validate?: Validator<Data, void, ValidationSource>;
+      status?: { expected: number | number[] };
+    };
+  }
+): Mutation<void, Data, ApiRequestError | Error | InvalidDataError>;
+
+// -- Implementation --
+
+function createJsonMutation(config: any): Mutation<any, any, any> {
+  const requestFx = createJsonApiRequest({
+    request: { method: config.request.method, credentials: 'same-origin' },
```

Comment:
Shouldn't `credentials` also be configurable? 🤔 

E.g. one may need to set it to `include` or `omit` for some reason

Reply:
It could, but let's wait for the real cases. Now, I would prefer reasonable default.